src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -25,25 +25,38 @@ public override async Task RunAsync(CancellationToken ct)
             // Reset previous checkpoints corresponding to an older source event hub (i.e. applicable if the source event hub changes)
             await CheckpointClient.ResetCheckpointsAsync(ct);
 
-            EventProcessorClient.ProcessEventAsync += ProcessEventHandler;
-            EventProcessorClient.ProcessErrorAsync += ProcessErrorHandler;
-            EventProcessorClient.PartitionInitializingAsync += ProcessInitializingHandler;
+            bool connected = false;
 
-            try
+            while (!connected)
             {
-                Console.WriteLine($"Starting event hub processor at {DateTime.UtcNow}");
-                await EventProcessorClient.StartProcessingAsync(ct);
+                try
+                {
+                    EventProcessorClient.ProcessEventAsync += ProcessEventHandler;
```

Comment:
Suggestion: Do we need to remove/add the event handlers when there is an exception? Or can we do the loop just around the start/wait and and have a try/catch just for ownership failures?

Reply:
we can add/remove the handlers outside of the while loop. updated the code to add/remove outside of the while loop